### PR TITLE
Implement support for parsing php 7.4 typed properties

### LINKED ISSUE #276: PHP 7.4-dev typed property support
See https://wiki.php.net/rfc/typed_properties_v2#proposal , which was merged yesterday

The following change was made to php-src's parser implementation

```patch
--- a/Zend/zend_language_parser.y
+++ b/Zend/zend_language_parser.y
@@ -716,8 +716,9 @@ class_statement_list:
 
 
 class_statement:
-               variable_modifiers property_list ';'
-                       { $$ = $2; $$->attr = $1; }
+               variable_modifiers optional_type property_list ';'
+                       { $$ = zend_ast_create(ZEND_AST_PROP_GROUP, $2, $3);
+                         $$->attr = $1; }
```

So the below change should hopefully work with corresponding changes to PropertyDeclaration (void is invalid for properties, like params):

```patch
--- a/src/Parser.php
+++ b/src/Parser.php
@@ -2776,6 +2776,8 @@ class Parser {
         $propertyDeclaration->parent = $parentNode;
 
         $propertyDeclaration->modifiers = $modifiers;
+        $propertyDeclaration->questionToken = $this->eatOptional1(TokenKind::QuestionToken);
+        $propertyDeclaration->type = $this->parseParamTypeDeclaration($propertyDeclaration);
         $propertyDeclaration->propertyElements = $this->parseExpressionList($propertyDeclaration);
         $propertyDeclaration->semicolon = $this->eat1(TokenKind::SemicolonToken);
```

### PR DESCRIPTION
**Pull request by TysonAndre**
Add tests of typed properties and tolerating errors in edge cases.

To maintain invariants
(and to avoid creating PropertyDeclaration without a name),
the questionToken and type were added to MissingMemberDeclaration.

Fixes #276

### CODE REVIEW

**Comment by roblourens** [comment]
Awesome, thanks! Sorry it took me so long to get to it.
